File: report/src/main/java/com/vmlens/report/element/TestResult.java
package com.vmlens.report.element;

import com.vmlens.report.description.DescriptionContext;
import com.vmlens.report.description.NeedsDescriptionCallback;
import com.vmlens.report.uielement.UIWarning;

import java.util.LinkedList;
import java.util.List;

public class TestResult {
    
    private String text;
    private final List<LargeWarning> largeWarningList = new LinkedList<>();
    
    public TestResult(String text) {
        this.text = text;
    }

    public String text() {
        return text;
    }

    public void addSmallWarningText(String warning) {
        text += ", " +  warning;
    }

    public void addLargeWarning(LargeWarning largeWarning) {
        largeWarningList.add(largeWarning);
    }

    public void addToNeedsDescription(NeedsDescriptionCallback callback) {
        for(LargeWarning warning : largeWarningList) {
            warning.addToNeedsDescription(callback);
        }
    }

    public List<UIWarning> toUIWarningList(DescriptionContext descriptionContext) {
        List<UIWarning> result = new LinkedList<>();
        for(LargeWarning warning : largeWarningList) {
            result.add(warning.toUIWarning(descriptionContext));
        }
        return result;
    }


}

File: sync-bug/src/main/scala/com/anarsoft/race/detection/report/element/TestResult.scala
package com.anarsoft.race.detection.report.element

import com.vmlens.report.overview.UIWarning

import java.util
import scala.collection.mutable

class TestResult(var text: String) {

  val largeWarningList: java.util.List[UIWarning] = new util.LinkedList[UIWarning]
  
  def addSmallWarningText(warning: String): Unit = {
    text += ", " + warning
  }

  def addLargeWarning(largeWarning: UIWarning): Unit = {
    largeWarningList.add(largeWarning)
  }
  
}
